pandas/io/parsers.py

```diff
@@ -976,17 +978,26 @@ def _is_index_col(col):
     return col is not None and col is not False
 
 
+def _evaluate_usecols(usecols, names):
+    if callable(usecols):
+        return set([i for i, name in enumerate(names)
+                    if com._apply_if_callable(usecols, name)])
```

Comment:
I think the `com._apply_if_callable` is not needed anymore as you already did `if callable(usecols)`? (apply_if_callable does exactly the same under the hood)